firebase-perf/src/main/java/com/google/firebase/perf/application/FrameMetricsRecorder.java

```diff
@@ -111,6 +111,12 @@ public Optional<PerfFrameMetrics> stop() {
     } catch (IllegalArgumentException err) {
       logger.warn(
           "View not hardware accelerated. Unable to collect FrameMetrics. %s", err.toString());
+      frameMetricsAggregator.reset();
+      return Optional.absent();
+    } catch (NullPointerException err) {
+      frameMetricsAggregator.reset();
+      logger.warn(
+          "NullPointerException. %s", err.toString());
```

Comment:
nit: Does the err.toString() have enough context to understand where the issue came from? Otherwise you can be more clear and say "NullPointerException when calling FrameMetricsAggregator.remove()"

Reply:
Recategorized the error, please see updated description